src/data_structure/check_data_structure.jl

```diff
@@ -91,13 +98,33 @@ Check that each `node` has at least one `temporal_block` connected to it in each
 function check_model__node__temporal_block()
     errors = [
         (m, n) for m in model(model_type=:spineopt_operations)
-        for n in node() if isempty(intersect(node__temporal_block(node=n), model__temporal_block(model=m)))
+        for n in node() if isempty(intersect(node__temporal_block(node=n), model__temporal_block(model=m))) && n == members(n)
     ]
     _check(
         isempty(errors),
         "invalid `node__temporal_block` or `model__temporal_block` definitions for `(model, node)` pair(s):
         $(join(errors, ", ", " and ")) " * "- each `node` must be related to at least one `temporal_block` per `model`",
     )
+    error_group = [
+        (m, n) for m in model(model_type=:spineopt_operations)
+        for n in node() if isempty(intersect(node__temporal_block(node=members(n)), model__temporal_block(model=m)))# && n != members(n)
+    ]
+    _check(
+        isempty(error_group),
+        "Some nodes in the node groups don't have a `node__temporal_block` or `model__temporal_block` definitions for `(model, node)` pair(s):
+        $(join(error_group, ", ", " and ")) " *
+        "- each `node` of the node groups must be related to at least one `temporal_block` per `model`",
+    )
+    warning = [
```

Comment:
Maybe `warnings` instead of `warning`? To be consistent with `errors` (plural) above...